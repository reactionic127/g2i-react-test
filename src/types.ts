export type SelectType = {
  value: string;
  label: string;
  key: string;
};

export type ModalType = {
  visible: boolean;
  setVisible: (args: boolean) => void;
};

export type ListType = {
  activities: ActivityType[];
};

export type ActivityType = {
  type: string;
  url?: string;
  note: string;
};

export type ActivityState = {
  activities: ActivityType[];
};

export type RootState = {
  activityState: ActivityState;
};

export type PayloadType = {
  type: string;
  payload: {
    type: string;
    url?: string;
    note: string;
  };
};
