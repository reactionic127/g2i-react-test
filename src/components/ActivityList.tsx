import React, { ReactElement } from "react";
import { List, Typography, Avatar } from "antd";
import { ListType } from "../types";

const currentUser = {
  name: "Richard Gillbert",
  shortname: "RG",
};

function ActivityList({ activities }: ListType): ReactElement {
  return (
    <List
      data-id="list"
      style={{ marginTop: 20, maxHeight: 300, overflowY: "auto" }}
      bordered
      dataSource={activities}
      renderItem={(item) => {
        return (
          <List.Item data-id="list-item">
            <Avatar
              style={{ backgroundColor: "#F0685E", marginRight: 10 }}
              size={40}
            >
              {currentUser.shortname}
            </Avatar>
            <Typography.Text strong>
              {currentUser.name} added a {item.type}:&nbsp;
            </Typography.Text>
            {item.url} {item.note}
          </List.Item>
        );
      }}
    />
  );
}

export default ActivityList;
