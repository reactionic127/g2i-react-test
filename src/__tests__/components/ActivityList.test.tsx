import React from "react";
import { shallow, ShallowWrapper } from "enzyme";
import ActivityList from "../../components/ActivityList";
import { ActivityType } from "../../types";

const mockDispatch = jest.fn();
jest.mock("react-redux", () => ({
  useSelector: () => ({ activities: [] }),
  useDispatch: () => mockDispatch,
}));

describe("ActivityList", () => {
  describe("rendering", () => {
    let wrapper: ShallowWrapper;
    const activities: ActivityType[] = [
      {
        type: "test type",
        url: "test url",
        note: "test note",
      },
    ];
    beforeEach(() => {
      wrapper = shallow(<ActivityList activities={activities} />);
    });

    it("renders the component", () => {
      expect(wrapper).toMatchSnapshot();
    });

    it("should have List", () => {
      expect(wrapper.find('[data-id="list"]').length).toEqual(1);
    });
  });
});
