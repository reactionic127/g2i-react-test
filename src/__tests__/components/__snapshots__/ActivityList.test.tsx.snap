// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActivityList rendering renders the component 1`] = `
<List
  bordered={true}
  data-id="list"
  dataSource={
    Array [
      Object {
        "note": "test note",
        "type": "test type",
        "url": "test url",
      },
    ]
  }
  renderItem={[Function]}
  style={
    Object {
      "marginTop": 20,
      "maxHeight": 300,
      "overflowY": "auto",
    }
  }
/>
`;
